addons/website_sale/controllers/main.py

```diff
@@ -1327,18 +1327,21 @@ def address(self, **kw):
                             kw['callback'] = '/shop/address'
                     elif address_mode == 'shipping':
                         update_values['partner_shipping_id'] = partner_id
+                        update_values['access_point_address'] = {}
                 elif update_mode == 'edit':  # Updating an existing address
                     if order.partner_id.id == partner_id:
                         # Editing the main partner of the SO --> also trigger a partner update to
                         # recompute fpos & any partner-related fields
                         update_values['partner_id'] = partner_id
+                        update_values['access_point_address'] = {}
 
                     if address_mode == 'billing':
                         update_values['partner_invoice_id'] = partner_id
                         if not kw.get('callback') and not order.only_services:
                             kw['callback'] = '/shop/checkout'
                     elif address_mode == 'shipping':
                         update_values['partner_shipping_id'] = partner_id
+                        update_values['access_point_address'] = {}
```

Comment:
When creating a new shipping address or changing one access point would not be reset as we are not going in `/shop/cart/update_address` route but instead `/shop/address`, as such we also need to reset `access_point_address` here too.

Reply:
honestly, it is becoming  difficult to maintain, maybe it is better to override `_remove_delivery_line()` of `sale.order` in `website_sale` module and reset pickup point when the carrier is removed.

Reply:
Hello, I tried this approach but it didn't work as ` is only triggered when changing shipping methods not address.
I removed this fix as it's actually not that big of a deal since pay button is disabled which means that we can't have a weird state where address is in one place and access point belongs to a different city.